x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/maintenance_window_callout/maintenance_window_callout.test.tsx

```diff
@@ -0,0 +1,223 @@
+    { gte: '2024-04-19T10:36:14.028Z', lte: '2024-04-19T10:37:00.000Z' },
+  ],
+  id: '5eafe070-e030-11ed-ac81-bfcb4982b4fd',
+  status: 'upcoming',
+  expiration_date: '2024-04-21T10:36:26.999Z',
+  r_rule: {
+    dtstart: '2023-04-21T10:36:14.028Z',
+    tzid: 'Europe/Amsterdam',
+    freq: 3,
+    interval: 1,
+    byweekday: ['FR'],
+  },
+  created_by: 'elastic',
+  updated_by: 'elastic',
+  created_at: '2023-04-21T10:36:26.999Z',
+  updated_at: '2023-04-21T10:36:26.999Z',
+  event_start_time: '2023-04-28T10:36:14.028Z',
+  event_end_time: '2023-04-28T10:37:00.000Z',
+};
+
+describe('MaintenanceWindowCallout', () => {
+  let appToastsMock: jest.Mocked<ReturnType<typeof useAppToastsMock.create>>;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    appToastsMock = useAppToastsMock.create();
+    (useAppToasts as jest.Mock).mockReturnValue(appToastsMock);
+  });
+
+  afterEach(() => {
+    cleanup();
+    jest.restoreAllMocks();
+  });
+
+  it('should be visible if currently there is at least one "running" maintenance window', async () => {
+    (fetchActiveMaintenanceWindows as jest.Mock).mockResolvedValue([RUNNING_MAINTENANCE_WINDOW_1]);
+
+    const { findByText } = render(<MaintenanceWindowCallout />, { wrapper: TestProviders });
+
+    expect(await findByText('A maintenance window is currently running')).toBeInTheDocument();
+  });
+
+  it('single callout should be visible if currently there are multiple "running" maintenance windows', async () => {
+    (fetchActiveMaintenanceWindows as jest.Mock).mockResolvedValue([
+      RUNNING_MAINTENANCE_WINDOW_1,
+      RUNNING_MAINTENANCE_WINDOW_2,
+    ]);
+
+    const { findAllByText } = render(<MaintenanceWindowCallout />, { wrapper: TestProviders });
+
+    expect(await findAllByText('A maintenance window is currently running')).toHaveLength(1);
+  });
+
+  it('should NOT be visible if currently there are no active (running or upcoming) maintenance windows', async () => {
+    (fetchActiveMaintenanceWindows as jest.Mock).mockResolvedValue([]);
+
+    const { container } = render(<MaintenanceWindowCallout />, { wrapper: TestProviders });
+
+    expect(container).toBeEmptyDOMElement();
+  });
+
+  it('should NOT be visible if currently there are no "running" maintenance windows', async () => {
+    (fetchActiveMaintenanceWindows as jest.Mock).mockResolvedValue([UPCOMING_MAINTENANCE_WINDOW]);
+
+    const { container } = render(<MaintenanceWindowCallout />, { wrapper: TestProviders });
+
+    expect(container).toBeEmptyDOMElement();
+  });
+
+  it('should see an error toast if there was an error while fetching maintenance windows', async () => {
+    const createReactQueryWrapper = () => {
+      const queryClient = new QueryClient({
+        defaultOptions: {
+          queries: {
+            // Turn retries off, otherwise we won't be able to test errors
+            retry: false,
+          },
+        },
+      });
```

Comment:
I remember your question about errors in the jest logs, I think there's a way to suppress react-query logging in tests:

https://stackoverflow.com/questions/65570462/how-do-i-mute-react-query-console-errors-when-a-promise-rejects

Not sure if there would be any non-obvious side-effects to this.